src/test/java/io/cdap/plugin/gcp/bigquery/BigQueryRecordToJsonTest.java

```diff
@@ -225,4 +241,220 @@ public void testNullableRecord() throws IOException {
+                                     recordField.getSchema(), jsonStringFieldsPaths);
+        }
+      }
+      writer.endObject();
+
+      JsonObject actual = writer.get().getAsJsonObject();
+
+      Assert.assertEquals(jsonObject, actual.get("jsonString").getAsJsonObject());
+    } catch (IOException e) {
+      Assert.fail("Failed to convert record to json");
+    }
+  }
+
+  @Test
+  public void testJsonStringWithEmptyObject() {
+    Schema recordSchema = Schema.recordOf(
+            "record",
+            Schema.Field.of("jsonString", Schema.of(Schema.Type.STRING))
+    );
+    String jsonString = "{}";
+    JsonObject jsonObject = gson.fromJson(jsonString, JsonObject.class);
+    StructuredRecord record = StructuredRecord.builder(recordSchema).set("jsonString", jsonString).build();
+    Set<String> jsonStringFieldsPaths = ImmutableSet.of("jsonString");
+    try {
+      JsonTreeWriter writer = new JsonTreeWriter();
+      writer.beginObject();
+      for (Schema.Field recordField : Objects.requireNonNull(record.getSchema().getFields())) {
+        if (recordSchema.getField(recordField.getName()) != null) {
+          BigQueryRecordToJson.write(writer, recordField.getName(), record.get(recordField.getName()),
+                                     recordField.getSchema(), jsonStringFieldsPaths);
+        }
+      }
+      writer.endObject();
+
+      JsonObject actual = writer.get().getAsJsonObject();
+
+      Assert.assertEquals(jsonObject, actual.get("jsonString").getAsJsonObject());
+    } catch (IOException e) {
+      Assert.fail("Failed to convert record to json");
+    }
+  }
+
+  @Test
+  public void testJsonStringWithEmptyArray() {
+    Schema recordSchema = Schema.recordOf(
+            "record",
+            Schema.Field.of("jsonString", Schema.of(Schema.Type.STRING))
+    );
+    String jsonString = "[]";
+    JsonArray jsonObject = gson.fromJson(jsonString, JsonArray.class);
+    StructuredRecord record = StructuredRecord.builder(recordSchema).set("jsonString", jsonString).build();
+    Set<String> jsonStringFieldsPaths = ImmutableSet.of("jsonString");
+    try {
+      JsonTreeWriter writer = new JsonTreeWriter();
+      writer.beginObject();
+      for (Schema.Field recordField : Objects.requireNonNull(record.getSchema().getFields())) {
+        if (recordSchema.getField(recordField.getName()) != null) {
+          BigQueryRecordToJson.write(writer, recordField.getName(), record.get(recordField.getName()),
+                                     recordField.getSchema(), jsonStringFieldsPaths);
+        }
+      }
+      writer.endObject();
+
+      JsonObject actual = writer.get().getAsJsonObject();
+
+      Assert.assertEquals(jsonObject, actual.get("jsonString").getAsJsonArray());
+    } catch (IOException e) {
+      Assert.fail("Failed to convert record to json");
+    }
+  }
+
+
+  @Test(expected = IllegalStateException.class)
+  public void testEmptyJsonString() throws IOException {
```

Comment:
Please mention the reason as a comment, why this exception is expected?

Reply:
```java
  /**
   * Empty JSON string is not a valid JSON string and should throw an exception.
   * @throws IOException
   */
   ```